pymatviz/ptable.py

```diff
@@ -1039,238 +1395,221 @@ def ptable_plots(
-                xycoords="axes fraction",
-                fontsize=8,
-                horizontalalignment="center",
-                verticalalignment="center",
-            )
-            if callable(anno_kwds):
-                annotation = anno_kwds(plot_data)
-            else:
-                annotation = anno_kwds
-                anno_text = anno_kwds.get("text")
-                if isinstance(anno_text, dict):
-                    anno_text = anno_text.get(symbol)
-                elif callable(anno_text):
-                    anno_text = anno_text(plot_data)
-                annotation["text"] = anno_text
-            ax.annotate(**(defaults | annotation))
+    child_args = child_args or {}
 
-        if plot_data is not None:
-            if callable(plot_kwds):
-                scatter_kwargs = plot_kwds(plot_data)
-            else:
-                scatter_kwargs = plot_kwds or {}
-
-            # Plot without colormap when data len is 2
-            if plot_data.ndim == 2:
-                ax.plot(plot_data[0], plot_data[1], **(scatter_kwargs or {}))
-
-            # Plot with colormap when data len is 3
-            elif plot_data.ndim == 3:
-                ax.plot(
-                    plot_data[0],
-                    plot_data[1],
-                    c=plot_data[2],
-                    cmap=cmap,
-                    **(scatter_kwargs or {}),
-                )
+    symbol_kwargs = symbol_kwargs or {}
+    symbol_kwargs.setdefault("fontsize", 12)
 
-            ax.tick_params(labelsize=8, direction="out")
-            if ax_kwds:
-                ax.set(**ax_kwds(plot_data) if callable(ax_kwds) else ax_kwds)
+    # Initialize periodic table plotter
+    plotter = PTableProjector(
+        data=data,
+        colormap=None,
+        plot_kwargs=plot_kwargs,  # type: ignore[arg-type]
+    )
 
-        # Disable ticks for elements without data
-        else:
-            ax.set_xticks([])
-            ax.set_yticks([])
+    # Call child plotter: line
+    plotter.add_child_plots(
+        ChildPlotters.line,
+        child_args=child_args,
+        ax_kwargs=ax_kwargs,
+        on_empty=on_empty,
+    )
 
-        # Hide right/top borders
-        for side in ("right", "top"):
-            ax.spines[side].set_visible(b=False)
+    # Add element symbols
+    plotter.add_ele_symbols(
+        text=symbol_text,  # type: ignore[arg-type]
+        pos=symbol_pos,
+        kwargs=symbol_kwargs,
+    )
 
-    # Add colorbar if data dimensionality is 3
-    if isinstance(cmap, Colormap) and {len(data) for data in data.values()} == {3}:
-        # Get the min/max values for norm calculation
-        third_lists = [data[2] for data in data.values()]
-        vmin = min(min(third_list) for third_list in third_lists)
-        vmax = max(max(third_list) for third_list in third_lists)
+    return plotter.fig
 
-        norm = Normalize(vmin=vmin, vmax=vmax)
 
-        cbar_ax = fig.add_axes(cbar_coords)
+def ptable_splits(
```

Comment:
maybe rename this to `ptable_heatmap_splits` so users immediately know what is being split?

Reply:
Yes! Such a wonderful idea.